### PR TITLE
parameterized spark compatible build

### PR DESCRIPTION
**Pull request by dliubimov**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -87,7 +94,11 @@ publishing {
     publications {
         maven(MavenPublication) {
             groupId = 'org.opendatadiscovery'
-            artifactId = 'oddrn-generator-java'
+            if (spark_compatible) {
+                artifactId = 'oddrn-generator-java8'
+            } else {
+                artifactId = 'oddrn-generator-java'
```
Comment (DementevNikita) [inline build.gradle:100]: @dliubimov
should we make it 'oddrn-generator-java11' then?
what's the best practice for naming here?

File: build.gradle [hunk 2]
```diff
@@ -9,15 +9,22 @@ plugins {
 group = 'org.opendatadiscovery'
 version = file('VERSION').text
 
-java {
-    sourceCompatibility = JavaVersion.VERSION_11
-    targetCompatibility = JavaVersion.VERSION_11
-}
-
 ext {
+    spark_compatible = findProperty('spark_compatible') ?: false
```
Comment (germanosin) [inline build.gradle:13]: let's name it java8 compatibale)

**Review by germanosin — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (dliubimov) [reply build.gradle]: done

**Comment by DementevNikita** [comment]
@dliubimov 

it also would be great if you could fix the linter errors (`gradle checkstyleMain`)

**Comment by dliubimov** [comment]
> 

@DementevNikita done

**Review by DementevNikita — APPROVED** [review]

**Review by germanosin — APPROVED** [review]

**Pull request merged**